apisix/init.lua

```diff
@@ -267,6 +267,17 @@ local function set_upstream_headers(api_ctx, picked_server)
         api_ctx.var.var_x_forwarded_host = x_forwarded_host
     end
 
+    local x_forwarded_for = api_ctx.var.http_x_forwarded_for
```

Comment:
Would be better to use http://nginx.org/en/docs/http/ngx_http_proxy_module.html#var_proxy_add_x_forwarded_for directly?

Reply:
Done